packages/pwa-kit-runtime/src/utils/loggerFactory.js

```diff
@@ -0,0 +1,132 @@
+            finalNamespace = namespace
+        }
+
+        return `${finalNamespace} ${level.toUpperCase()} ${message}${
+            additionalProperties ? ` ${JSON.stringify(additionalProperties)}` : ''
+        }`
+    }
+
+    /**
+     * Prints a log message with the namespace using the console object method set in the message log level.
+     * @param {string} message - The log message.
+     * @param {Object} options - Optional message details.
+     * @param {string} options.level - The log level of the message.
+     */
+    #printLog(message, options = {}) {
+        const {level} = options
+
+        const formattedMessage = this.#formatLogMessage(message, options)
+
+        switch (level) {
+            case 'error':
+                console.error(formattedMessage)
+                break
+            case 'warn':
+                console.warn(formattedMessage)
+                break
+            case 'info':
+                console.info(formattedMessage)
+                break
+            case 'debug':
+                console.debug(formattedMessage)
+                break
+            default:
+                console.log(formattedMessage)
+                break
+        }
+    }
+
+    /**
+     * Logs a debug message.
+     * @param {string} message - The debug message.
+     * @param {Object} [details={}] - Optional message details.
+     */
+    debug(message, details = {}) {
+        this.#printLog(message, {level: 'debug', ...details})
+    }
+
+    /**
+     * Logs an info message.
+     * @param {string} message - The info message.
+     * @param {Object} [details={}] - Optional message details.
+     */
+    log(message, details = {}) {
+        this.#printLog(message, {level: 'info', ...details})
+    }
+
+    /**
+     * Logs an info message.
+     * @param {string} message - The info message.
+     * @param {Object} [details={}] - Optional message details.
+     */
+    info(message, details = {}) {
+        this.#printLog(message, {level: 'info', ...details})
+    }
+
+    /**
+     * Logs a warning message.
+     * @param {string} message - The warning message.
+     * @param {Object} [details={}] - Optional message details.
+     */
+    warn(message, details = {}) {
+        this.#printLog(message, {level: 'warn', ...details})
+    }
+
+    /**
+     * Logs an error message.
+     * @param {string} message - The error message.
+     * @param {Object} [details={}] - Optional message details.
+     */
+    error(message, details = {}) {
+        this.#printLog(message, {level: 'error', ...details})
+    }
+}
+
+/**
+ * Create a logger instance for each package.
+ *
+ * @param {Object} config - Configuration object for the logger.
+ * @param {string} config.packageName - The name of the package where the logger is used.
+ * @returns {PWAKITLogger} - An instance of PWAKITLogger configured for the specified package.
+ */
+const createLogger = (config = {}) => {
+    return new PWAKITLogger(config)
```

Comment:
Here you are using the Singleton pattern, from you might want to freeze `new PWAKitLogger(config)` with `Object.freeze()` so is cannot be modified